src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,49 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * Handles delete command, which deletes the specified task from the list of tasks.
+ */
+public class DeleteCommand extends Command {
+    /**
+     * Index of task to be deleted.
+     */
+    private final int index;
+
+    /**
+     * @param index index of task to be deleted
+     */
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Delete task at the index from the list of tasks. This method also prints out the result to the interface.
+     *
+     * @param tasks   list of all the tasks
+     * @param ui      interface to print to
+     * @param storage storage to save the tasks
+     * @throws DukeException If fails to delete the task at the index, or fails to save to the local storage.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
Correct me if I am wrong. The delete command only deletes a task based on the index of the task on the whole tasks list. However, this application also has the Find Command. When performing the find, the found tasks are also listed in the same way as the list command. If the user wants to delete tasks in the found tasks with an index, the application might go wrong by deleting the task with the wrong index.

Reply:
You might want to add a variable for keep tracking the last shown list of tasks to the user. As we discussed in the last tutorial (the PersonBook#lastShownList variable)